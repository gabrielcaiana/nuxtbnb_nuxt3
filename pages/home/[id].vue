<script setup>
const route = useRoute();
const nuxtApp = useNuxtApp();
const algolia = useAlgolia();

const map = ref(null);
const { home, pending, error } = await algolia.getHomes(route.params.id);

if (error.value) {
  throw new Error(error.value);
}

onMounted(() => {
  setTimeout(() => {
    nuxtApp.$showMap(map.value, home.value._geoloc.lat, home.value._geoloc.lng);
  }, 250);
});
</script>

<template>
  <div>
    <div v-if="!pending">
      <Head>
        <Title>{{ home.title }}</Title>
      </Head>

      <div class="flex">
        <img
          v-for="(image, index) in home.images"
          :key="index"
          :src="image"
          :alt="home.title"
          width="200"
          height="150"
        />
      </div>
      {{ home.title }} <br />
      {{ home.pricePerNight }} <br />
      <img src="/marker.svg" width="20" height="20" alt="marker" />
      {{ home.location.address }}
      {{ home.location.state }} {{ home.location.country }} <br />
      <img src="/star.svg" width="20" height="20" alt="star" />
      {{ home.reviewValue }} {{ home.guests }} guests,
      {{ home.bedrooms }} rooms, {{ home.beds }}, {{ home.bathrooms }} bath
      <br />

      <!-- init map -->
      <div class="w-[800px] h-[800px]" ref="map"></div>
    </div>
    <div v-else>Carregando...</div>
  </div>
</template>
