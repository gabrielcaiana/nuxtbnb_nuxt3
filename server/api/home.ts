import { serverSupabaseClient } from "#supabase/server";

export default defineEventHandler(async (event) => {
  const client = serverSupabaseClient(event);
  const query = getQuery(event);

  const { data } = await client.from("homes").select().eq("id", query.id);

  if (!data?.length) {
    throw createError({ statusCode: 404, statusMessage: "Page Not Found" });
  }

  return { data };
});
