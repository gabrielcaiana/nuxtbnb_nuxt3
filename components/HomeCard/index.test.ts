import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import HomeCard from "./index.vue";
import home from "../../mocks/api/home";

describe("HomeCard.vue", () => {
  it("should render the correct component", () => {
    const wrapper = mount(HomeCard, {
      props: {
        home,
      },
    });

    const component = wrapper.find('[datatest-id="home-card"]');
    expect(component.exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
