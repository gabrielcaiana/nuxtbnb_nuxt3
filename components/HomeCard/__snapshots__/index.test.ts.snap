// Vitest Snapshot v1

exports[`HomeCard.vue > should render the correct component 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <div
    datatest-id="home-card"
  >
    <img
      alt="Rustic Piney Lodge Nestled in the Foothills"
      src="https://images.unsplash.com/photo-1514774849407-0bef2676db20?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1267&q=80"
      style="width: 200px;"
    />
     
    <br />
     Rustic Piney Lodge Nestled in the Foothills 
    <br />
     521 E Lionshead Circle Vail Colorado
    <br />
     4 guests 
    <br />
     219 / night 
    <br />
  </div>,
}
`;
